src/chocolatey.resources/helpers/functions/Get-Installed.ps1

```diff
@@ -0,0 +1,48 @@
+﻿## Get-Installed
+##############################################################################################################
+## Returns a list of installed applications.
+## Registry based replacement for "Get-WmiObject -Class Win32_Product".
+##############################################################################################################
+
+function Get-Installed {
+param(
+  [switch] $GWMICompatible = $false # You may use '-GWMICompatible' switch
+)
+
+  $items = @()
+
+  if(Test-Path "HKLM:SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall") {
+    $items += Get-ChildItem "HKLM:SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall"
+  }
+  if(Test-Path "HKLM:SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall") {
+    $items += Get-ChildItem "HKLM:SOFTWARE\WOW6432Node\Microsoft\Windows\CurrentVersion\Uninstall"
+  }
+
+  $productList = @()
+
+  foreach ($app in $items)
+  {
+    $propitems = $app | Foreach-Object {Get-ItemProperty $_.PsPath}
+    $w32item = New-Object System.Object
+    $w32item | Add-Member -type NoteProperty -Name "IdentifyingNumber" -Value $app.PsChildname
+    $w32item | Add-Member -type NoteProperty -Name "Name"    -Value $propitems.DisplayName
+    $w32item | Add-Member -type NoteProperty -Name "Vendor"  -Value $propitems.Publisher
+    $w32item | Add-Member -type NoteProperty -Name "Version" -Value $propitems.DisplayVersion

```

Comment:
@ferventcoder - I have done code for this before - ran and was debugged for 14000 machines in 17 MUI language on Win XP and Win 7.

In that code I tried to reconcile win32_product with this information - you basically have to link it by product code which can be parsed out of the ARP data.  But for my purposes it looked like that road was going to be very bumpy.  For once think if there is a master MSI and child MSIs - more than one ARP item can carry the product code.  Sometimes the same was true when a setup.exe contains an MSI.  So for my more limited purposes I just parsed Product Code from ARP data and didn't bother with contacting MSI.

Here are some other challenges / issues I ran into and codeded around (with no connotations of what would be the appropriate approach for chocolatey):
- there is a third per-user location for uninstall.  Kindle reader is one thing I've found in it before.  Exist  check it and pull it into the record list if it exists.
- raw uninstall keys contain some hidden items - created a switch to include hidden items rather than have them in the list all the time
- raw uninstall keys contain parent / child relationships not normally shown in Programs and Features - created a switch to include them rather than retrieve all the time.
- raw uninstall keys can contain windows updates KB article updates that don't normall show in Programs and Features - created a switch
- occasionally got blank records when retrieving from this key - used a filter to drop them after retrieving all three locations.
- it is possible, but infrequent, for this code to choke when poorly formed uninstall keys have been laid down by a vendor.  There was some java dev kit that was laying down "size" in a bizzare format and it would choke.  Found others had reported it to the vendor.  We only had about 4 affected machines in 14,000 - so just corrected those.  (this is probably extremely rare - but when the code chokes on it - it's hard for the individual to figure out why)
- The code provides three derived properties for GUIDfromUninstallString, KBGUIDFromKeyName, KBIDFromKeyName

For KBs the DisplayName will be null.  For childitems ParentKeyName will be populated.  Many times the parent item accomplishes an uninstall of child items - this is also what the "show updates" checkbox in Programs and Features" uses.

So I agree with your suspicion that this code would need some work to not generate any errors on the millions of nodes running Chocolatey.  I agree that the code should be updated - but you probably want to get a prototype out there to flush out the above issues and any others and transition to the new function after some debug time has occurred in the real world.